tests/test_membership.py

```diff
@@ -0,0 +1,171 @@
+        body=to_json_string(sample_group), status=200)
+    r = vinyldns_client.get_group('123')
+    check_groups_are_same(sample_group, r)
+
+
+def test_delete_group(mocked_responses, vinyldns_client):
+    mocked_responses.add(
+        responses.DELETE, 'http://test.com/groups/123',
+        body=to_json_string(sample_group), status=200)
+    r = vinyldns_client.delete_group('123')
+    check_groups_are_same(sample_group, r)
+
+
+def test_list_my_groups(mocked_responses, vinyldns_client):
+    sample_list_groups = ListGroupsResponse([sample_group, sample_group2], 5, '*', 'start-from', 'next-id')
+    mocked_responses.add(
+        responses.GET, 'http://test.com/groups?groupNameFilter=*&startFrom=start-from&maxItems=5',
+        body=to_json_string(sample_list_groups), status=200)
+    r = vinyldns_client.list_my_groups('*', 'start-from', 5)
+    assert sample_list_groups.start_from == r.start_from
+    assert sample_list_groups.group_name_filter == r.group_name_filter
+    assert sample_list_groups.next_id == r.next_id
+
+    for l, r in zip(sample_list_groups.groups, r.groups):
+        check_groups_are_same(l, r)
+
+
+def test_list_all_my_groups(mocked_responses, vinyldns_client):
+    sample_list_groups1 = ListGroupsResponse([sample_group], 1, '*', 'start-from', 'next-id')
+
+    # Set next id to None to indicate end of list
+    sample_list_groups2 = ListGroupsResponse([sample_group2], 1, '*', 'start-from', next_id=None)
+    mocked_responses.add(
+        responses.GET, 'http://test.com/groups?groupNameFilter=*',
+        body=to_json_string(sample_list_groups1), status=200)
+
+    # list all puts the next id from the first response into the startFrom for the next request
+    mocked_responses.add(
+        responses.GET, 'http://test.com/groups?groupNameFilter=*&startFrom=next-id',
+        body=to_json_string(sample_list_groups2), status=200)
+    r = vinyldns_client.list_all_my_groups('*')
+
+    assert r.start_from is None
+    assert r.next_id is None
+    assert sample_list_groups1.group_name_filter == r.group_name_filter
+
+    for l, r in zip([sample_group, sample_group2], r.groups):
+        check_groups_are_same(l, r)
+
+
+def test_list_members(mocked_responses, vinyldns_client):
+    member1 = Member('some-id', 'user-name', 'first', 'last', 'test@test.com', datetime.datetime.utcnow(), False)
+    member2 = Member('some-id2', 'user-name2', 'first2', 'last2', 'test2@test.com', datetime.datetime.utcnow(), False)
+    lmr = ListMembersResponse([member1, member2], start_from='start', next_id='next', max_items=100)
+    mocked_responses.add(
+        responses.GET, 'http://test.com/groups/foo/members?startFrom=start&maxItems=100',
+        body=to_json_string(lmr), status=200
+    )
+    r = vinyldns_client.list_members_group('foo', 'start', 100)
+    r.start_from = lmr.start_from
+    r.next_id = lmr.next_id
+    r.max_items = lmr.max_items
+
+    for l, r in zip(lmr.members, r.members):
+        assert l.id == r.id
```

Comment:
A lot of the same assertions are called in `test_list_members` and `test_list_group_admins` - might be better to have a common function with the assertions that we can call from both.

Reply:
They are different data types so I didn't want to use the same assertions even though they are very similar